src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs

```diff
@@ -398,29 +398,96 @@ private static bool TryParseCommonFramework(string frameworkString, out NuGetFra
         {
             framework = null;
 
-            if (StringComparer.OrdinalIgnoreCase.Equals(frameworkString, "dotnet"))
-            {
-                framework = FrameworkConstants.CommonFrameworks.DotNet50;
-            }
-            else if (StringComparer.OrdinalIgnoreCase.Equals(frameworkString, "dnx")
-                     || StringComparer.OrdinalIgnoreCase.Equals(frameworkString, "dnx451"))
-            {
-                framework = FrameworkConstants.CommonFrameworks.Dnx451;
-            }
-            else if (StringComparer.OrdinalIgnoreCase.Equals(frameworkString, "dnxcore")
-                     || StringComparer.OrdinalIgnoreCase.Equals(frameworkString, "dnxcore50")
-                     || StringComparer.OrdinalIgnoreCase.Equals(frameworkString, "dnxcore5"))
-            {
-                framework = FrameworkConstants.CommonFrameworks.DnxCore50;
-            }
-            else if (StringComparer.OrdinalIgnoreCase.Equals(frameworkString, "net40")
-                     || StringComparer.OrdinalIgnoreCase.Equals(frameworkString, "net4"))
-            {
-                framework = FrameworkConstants.CommonFrameworks.Net4;
-            }
-            else if (StringComparer.OrdinalIgnoreCase.Equals(frameworkString, "net45"))
+            frameworkString = frameworkString.ToLowerInvariant();
+
+            switch (frameworkString)
```

Comment:
Just curious why we went with this route rather than a simple Dictionary<string, NuGetFramework> and generate them on the fly, but limit instances to one per unique string?